Sources/AsyncKit/ConnectionPool.swift

```diff
@@ -138,43 +139,67 @@ public final class ConnectionPool<Source> where Source: ConnectionPoolSource  {
     ///
     /// The connection returned by this method should be released when you are finished using it.
     ///
-    ///     let conn = try pool.requestConnection().wait()
+    ///     let conn = try pool.requestConnection(on: ...).wait()
     ///     defer { pool.releaseConnection(conn) }
     ///     // use the connection
     ///
     /// See `withConnection(_:)` for a callback-based method that automatically releases the connection.
     ///
+    /// - parameters:
+    ///     - on: Preferred event loop for the new connection.
     /// - returns: A future containing the requested connection.
-    public func requestConnection() -> EventLoopFuture<Source.Connection> {
-        guard !self.isClosed else {
-            return self.source.eventLoop.makeFailedFuture(ConnectionPoolError.closed)
+    public func requestConnection(on eventLoop: EventLoop) -> EventLoopFuture<Source.Connection> {
+        guard !self.didShutdown else {
+            return eventLoop.makeFailedFuture(ConnectionPoolError.shutdown)
         }
-        
+
+        // synchronize access to available / active connection checks
+        self.lock.lock()
+
         if let conn = self.available.popLast() {
```

Comment:
This can return a connection tied to a different event loop than the one passed into this method; is that expected?

Reply:
That part is incomplete. The idea is to have this be a "preferred" event loop. In the future more logic will be added to attempt to return the best possible connection pool given this information.